static/app/components/orgRole.tsx

```diff
@@ -0,0 +1,148 @@
+import {Fragment, useMemo} from 'react';
+import styled from '@emotion/styled';
+import * as Sentry from '@sentry/react';
+
+import ExternalLink from 'sentry/components/links/externalLink';
+import Link from 'sentry/components/links/link';
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconInfo} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Member, Organization} from 'sentry/types';
+import {getEffectiveOrgRole} from 'sentry/utils/orgRole';
+
+export const OrgRoleInfo = ({
+  organization,
+  member,
+}: {
+  member: Member;
+  organization: Organization;
+}) => {
+  const {orgRoleList} = organization;
+  const {orgRole, orgRolesFromTeams} = member;
+
+  const orgRoleFromMember = useMemo(() => {
+    const role = orgRoleList.find(r => r.id === orgRole);
+    return role;
+  }, [orgRole, orgRoleList]);
+
+  const effectiveOrgRole = useMemo(() => {
+    const memberOrgRoles = orgRolesFromTeams.map(r => r.role.id).concat([orgRole]);
+    return getEffectiveOrgRole(memberOrgRoles, orgRoleList);
+  }, [orgRole, orgRolesFromTeams, orgRoleList]);
+
+  if (!orgRoleFromMember) {
+    Sentry.withScope(scope => {
+      scope.setExtra('context', {
+        memberId: member.id,
+        orgRole: member.orgRole,
+      });
+      Sentry.captureException(new Error('OrgMember has an invalid orgRole.'));
+    });
```

Comment:
You probably want this to be in a `useEffect` right?

Reply:
There's no side-effects here. This component doesn't maintain state or change state.

Reply:
Sentry.captureExceotion is a side effect, we don’t want that firing when the component rerenders

Reply:
Oh yea, I understand now. Fixed.